compiler/rustc_typeck/src/check/pat.rs

```diff
@@ -1308,6 +1308,11 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 .copied()
                 .filter(|(field, _)| {
                     field.vis.is_accessible_from(tcx.parent_module(pat.hir_id).to_def_id(), tcx)
+                        && !matches!(
+                            tcx.eval_stability(field.did, None, DUMMY_SP, None),
+                            EvalResult::Deny { .. }
+                        )
+                        && !tcx.is_doc_hidden(field.did)
```

Comment:
Hmm, I don't think we should be hiding `doc(hidden)` fields in the same crate as their definition.